From ba70fa570d306242020b9f5d3d50af9ec73a08f0 Mon Sep 17 00:00:00 2001
From: Fugang Duan <B38611@freescale.com>
Date: Fri, 11 Nov 2011 11:36:21 +0800
Subject: [PATCH 0594/2463] ENGR00161617-1 [MX6] : Fix the build warning.

- Fix the building warning:
  common.h:75: warning: 'struct fec_platform_data' declared
  inside parameter list

  Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 arch/arm/mach-mx6/mx6_fec.c             |    2 +-
 arch/arm/plat-mxc/include/mach/common.h |    1 +
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx6/mx6_fec.c b/arch/arm/mach-mx6/mx6_fec.c
index 9d3c824..32e8583 100644
--- a/arch/arm/mach-mx6/mx6_fec.c
+++ b/arch/arm/mach-mx6/mx6_fec.c
@@ -43,7 +43,7 @@ static int fec_get_mac_addr(unsigned char *mac)
 	return 0;
 }
 
-void imx6_init_fec(struct fec_platform_data fec_data)
+void __init imx6_init_fec(struct fec_platform_data fec_data)
 {
 	fec_get_mac_addr(fec_data.mac);
 	if (!is_valid_ether_addr(fec_data.mac))
diff --git a/arch/arm/plat-mxc/include/mach/common.h b/arch/arm/plat-mxc/include/mach/common.h
index 1359c44..9a340cc 100755
--- a/arch/arm/plat-mxc/include/mach/common.h
+++ b/arch/arm/plat-mxc/include/mach/common.h
@@ -19,6 +19,7 @@
 #ifndef __ASM_ARCH_MXC_COMMON_H__
 #define __ASM_ARCH_MXC_COMMON_H__
 
+struct fec_platform_data;
 struct platform_device;
 struct clk;
 
-- 
1.7.7.4

